### PR TITLE
Additionally compile libmesh with dbg in Docker image

### LINKED ISSUE #18105: Moose basic create application using docker is failed
Hi,

I am using moose with docker. I tried to create application by using following steps in [Website](https://mooseframework.inl.gov/getting_started/new_users.html#create-an-app). When I run
```
./run_tests -j4
```
I am getting error message like "failed to import pyhit - try running make hit in $MOOSE_DIR/test"

How can I solve this issue?

### LINKED ISSUE #18924: Build libmesh dbg executable in Docker image
## Reason
For a standard development environment, it makes sense to be able to compile the dbg executable. Currently, the docker image only compiles libmesh in opt. dbg builds will fail. Encountered by @herter4171.

## Design
Build libmesh with dbg in Docker image.

## Impact
Improved Docker image.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #18924 and closes #18105

### CODE REVIEW

**Review by GiudGiud — DISMISSED** [review]

**Comment by GiudGiud** [comment]
is this Docker recipe broken?
It's not run for regular PRs isnt it

**Comment by dschwen** [comment]
Yeah, people are reporting problems with the docker image. #18105 

If u recall correctly the docker image was an external contribution.

**Comment by loganharbour** [comment]
@herter4171 in the process of turning off the docker tests due to timeouts, we let a failure slip in.

https://github.com/idaholab/moose/pull/18925/commits/b3ae07e1ff202d9cca4c96deb14ffb259d207d76 is the fix.

Added two new scripts for configure; hoping to get this fixed today

File: docker_ci/Dockerfile [hunk 1]
```diff
@@ -129,6 +132,7 @@ USER dev
 COPY --chown=dev:dev . ${MOOSE_DIR}
 
 ARG BUILD_MODULES
+ARG MOOSE_METHODS=opt
```
Comment (dschwen) [inline docker_ci/Dockerfile:135]: Is there a reason to have moose and libmesh methods separately? This leaves room for error as moose methods always have to be a subset of libmesh methods :-/

**Review by dschwen — APPROVED** [review]
Extra control for advanced users is fine. Defaults are sensible.

**Comment by loganharbour** [comment]
Merging despite the failed test - the failure is for an optional recipe and is for a WIP container.

**Pull request merged**